packages/effect/src/Effect.ts

```diff
@@ -94,6 +94,14 @@ export interface Effect<out A, out E = never, out R = never> extends Effect.Vari
   readonly [Unify.typeSymbol]?: unknown
   readonly [Unify.unifySymbol]?: EffectUnify<this>
   readonly [Unify.ignoreSymbol]?: EffectUnifyIgnore
+
+  [Symbol.iterator](): EffectGenerator<Effect<A, E, R>>
+}
```

Comment:
I'm not at a laptop to test, but maybe `this` in place of `Effect<A, E, R>` would work?

Reply:
I tried both and it didn't made a difference 🤷🏼‍♂️